src/compiler/scala/tools/nsc/ast/parser/Parsers.scala

```diff
@@ -829,7 +829,7 @@ self =>
       def mkNamed(args: List[Tree]) = if (isExpr) args map treeInfo.assignmentToMaybeNamedArg else args
       val arguments = right match {
         case Parens(args) => mkNamed(args)
-        case _            => List(right)
+        case _            => right :: Nil
```

Comment:
Yeah, we're lacking an optimization on `List.apply` to avoid boxing an array to pass as varargs. We've got a ticket open for that improvement, but in the meantime I'm happy enough with the readability of `::` to use it in the few places where this makes a material difference.